pkg/docs/reference.md

```diff
@@ -0,0 +1,582 @@
+# rules_pkg - reference
+
+<div class="toc">
+  <h2>Rules</h2>
+  <ul>
+    <li><a href="#pkg_tar">pkg_tar</a></li>
+    <li><a href="#pkg_zip">pkg_zip</a></li>
+    <li><a href="#pkg_deb">pkg_deb</a></li>
+    <li><a href="#pkg_rpm">pkg_rpm</a></li>
+  </ul>
+
+</div>
+
+<a name="pkg_tar"></a>
+## pkg_tar
+
+```python
+pkg_tar(name, extension, strip_prefix, package_dir, srcs,
+        mode, modes, deps, symlinks)
+```
+
+Creates a tar file from a list of inputs.
+
+<table class="table table-condensed table-bordered table-params">
+  <colgroup>
+    <col class="col-param" />
+    <col class="param-description" />
+  </colgroup>
+  <thead>
+    <tr>
+      <th colspan="2">Attributes</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><code>name</code></td>
+      <td>
+        <code>Name, required</code>
+        <p>A unique name for this rule.</p>
+      </td>
+    </tr>
+    <tr>
+      <td><code>extension</code></td>
+      <td>
+        <code>String, default to 'tar'</code>
+        <p>
+            The extension for the resulting tarball. The output
+            file will be '<i>name</i>.<i>extension</i>'. This extension
+            also decide on the compression: if set to <code>tar.gz</code>
+            or <code>tgz</code> then gzip compression will be used and
+            if set to <code>tar.bz2</code> or <code>tar.bzip2</code> then
+            bzip2 compression will be used.
```

Comment:
It would be nice if the formatting for this could use more markdown rather than HTML, but that can be left for a different day.

Reply:
I *think* stardoc can emit markdown, but I am not sure. In any case, I wanted to make this PR easier to review but just splitting the file and not modifying so much.